photon-ml/src/main/scala/com/linkedin/photon/ml/data/ScoredGameDatum.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data
+
+import scala.collection.Map
+
+import com.linkedin.photon.ml.util.Utils.isAlmostZero
+
+/**
+  * Representation of a single scored GAME data point
+  *
+  * @param response The response or label
+  * @param offset offset
+  * @param weight importance weight
+  * @param score computed score for this instance
+  * @param idTypeToValueMap The id type to value map that holds different types of ids associated with this data
+  *                         point. A few examples of the ids types are: (i) ids used to build the random effect model
+  *                         such as userId and itemId; (ii) ids used to compute certain metrics like precision@k such
+  *                         as documentId or queryId; (iii) ids that are used to uniquely identify each training record.
+  */
+case class ScoredGameDatum(response: Double, offset: Double, weight: Double,
+                           var score: Double, idTypeToValueMap: Map[String, String]) extends Serializable {
+  /**
+    * Get a copy of the current instance with a score of [[ScoredGameDatum.DEFAULT_SCORE]]
+    *
+    * @return copy of the current scored datum instance with zero score
+    */
+  def getZeroScoreDatum(): ScoredGameDatum = {
```

Comment:
Fine with me, as long as we are consistent. But if it is zero, yes, it's additive identity, so it can be something else. The problem is that the "zero" score datum is actually a vector, and only one element of that vector is going to be added (the score), and is going to be set to zero. There's some amount of built-in ambiguity.